### ISSUE TITLE
missing some phones in the generated TextGrid files [BUG]

### ISSUE THREAD

**Issue by zaynabmu** [MontrealCorpusTools/Montreal-Forced-Aligner#474]
**Debugging checklist**

[ ] Have you updated to latest MFA version? yes 2.0.1 
[ ] Have you tried rerunning the command with the `--clean` flag? yes 

**Describe the issue*

Actually I trained a new acoustic model on my language (Arabic)  correctly as documentation step by step , but  the generated TectGrid output contains (spn) and some phones were missing since I want to use TextGrids files in training Fastspeech model but the generated voices worse and then I figured out the reason from missing phones in TG files could you pls help me. 
![image](https://user-images.githubusercontent.com/89045234/174690393-6d3f3968-0494-4770-87c4-4585ab2a178c.png)

**For Reproducing your issue**
Please fill out the following:

1. Corpus structure
   * What language is the corpus in? Arabic 
   * How many files/speakers? 23/23
   * Are you using lab files or TextGrid files for input? lab files
2. Dictionary
   * Are you using a dictionary from MFA? If so, which one? no 
   * If it's a custom dictionary, what is the phoneset? Ara
3. Acoustic model
   * If you're using an acoustic model, is it one download through MFA? If so, which one? no
   * If it's a model you've trained, what data was it trained on? audio files with wav format 

**Log file**
Please attach the log file for the run that encountered an error (by default these will be stored in `~/Documents/MFA`).

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, OSX, Linux] Windows
 - Version [e.g. MacOSX 10.15, Ubuntu 20.04, Windows 10, etc] Windows 10
 - Any other details about the setup (Cloud, Docker, etc)

**Additional context**
when I run mfa validate everythings seems okay

**Comment by mmcauliffe** [comment]
The "spn" are generated from bracketed words or words not in the dictionary.  In the root temporary directory (~/Documents/MFA/{corpus_name}), there will be an "oovs_found.txt".  If you add those words to your dictionary with pronunciations, then they should show up correctly in the aligned TextGrid.

**Comment by peanut1101** [comment]
@mmcauliffe @zaynabmu I have encountered the same problem. Have you solved it? Please send me an email chen347444625@163.com

**Comment by zaynabmu** [comment]
> @mmcauliffe @zaynabmu I have encountered the same problem. Have you solved it? Please send me an email [chen347444625@163.com](mailto:chen347444625@163.com)

Yes its solved , follow the same instructions that mmcauliffe said and it will be work with you

**Comment by peanut1101** [comment]
@zaynabmu Thanks

**Comment by Lakhjeet1082** [comment]
> The "spn" are generated from bracketed words or words not in the dictionary. In the root temporary directory (~/Documents/MFA/{corpus_name}), there will be an "oovs_found.txt". If you add those words to your dictionary with pronunciations, then they should show up correctly in the aligned TextGrid.

i have added all oov words with their pronounciation to my dictionary and trained the acoustic model again for both the speakers, but still i am getting "spn" in the newly generated textgrid files. Please guide me on this

**Comment by chirila** [comment]
this also happens (I think) if the words contain segments that aren't in the phone set
